### PR TITLE
Release/snowplow-web/0.4.0

### PR DESCRIPTION
**Pull request by bill-warner**
TODO:
* Write up + link issues
* Add CHANGELOG

In the interest of time I thought I would start the PR now. I have run all tests in Redshift for dbt 0.18.2 and 0.19.2.

### CODE REVIEW

File: dbt_project.yml [hunk 1]
```diff
@@ -61,13 +64,15 @@ models:
...
+        +schema: "snowplow_manifest"
       scratch:
         +schema: "scratch"
         +tags: "scratch"
         bigquery:
           enabled: "{{ target.type == 'bigquery' | as_bool() }}"
-        redshift:
-          enabled: "{{ target.type == 'redshift' | as_bool() }}"
+        default:
+          enabled: "{{ target.type in ['redshift', 'postgres'] | as_bool() }}"
```
Comment (paulboocock) [inline dbt_project.yml:75]: Is this good practise? I appreciate Redshift and Postgres are the same right now... but they could deviate in the future, either at the warehouse level or the snowplow structure level. I fear we're setting ourselves up for needing a future fix by treating them as the same.

File: integration_tests/dbt_project.yml [hunk 2]
```diff
@@ -45,63 +45,63 @@ vars:
...
-          page_urlfragment: "{{ 'varchar' if target.type == 'redshift' else 'string' }}" 
-          refr_urlfragment: "{{ 'varchar' if target.type == 'redshift' else 'string' }}" 
-          refr_medium: "{{ 'varchar' if target.type == 'redshift' else 'string' }}" 
-          refr_source: "{{ 'varchar' if target.type == 'redshift' else 'string' }}" 
-          refr_term: "{{ 'varchar' if target.type == 'redshift' else 'string' }}" 
-          mkt_term: "{{ 'varchar' if target.type == 'redshift' else 'string' }}" 
-          mkt_content: "{{ 'varchar' if target.type == 'redshift' else 'string' }}" 
-          se_label: "{{ 'varchar' if target.type == 'redshift' else 'string' }}" 
-          se_property: "{{ 'varchar' if target.type == 'redshift' else 'string' }}" 
+          user_fingerprint: "{{ 'string' if target.type == 'bigquery' else 'varchar' }}"
```
Comment (paulboocock) [inline integration_tests/dbt_project.yml:56]: Does this change not change snowflake from `string` in 0.3.1 to `varchar` in 0.4.0? Is that ok?

**Review by paulboocock — APPROVED** [review]
LGTM. Couple of comments, likely due to my naivety rather than them being actual issues.

File: integration_tests/dbt_project.yml (hunk 2 above)
Comment (bill-warner) [reply integration_tests/dbt_project.yml:56]: Yeh you are correct it does change it but thankfully for Snowflake `string` and `varchar` are [synonymous](https://docs.snowflake.com/en/sql-reference/data-types-text.html#string-text). It is only BigQuery that requires `string` datatype.

File: dbt_project.yml (hunk 1 above)
Comment (bill-warner) [reply dbt_project.yml:75]: Fair point. My thinking here:
* If Postgres changes at a warehouse level then dbt should be handling a lot of those changes under the hood e.g. changes in DDL syntax etc.
* If the Snowplow structure changes then we can split out the Redshift and Postges models where appropriate. Throughout the package I have tried to consolidate the models across the various warehouses where possible in order to remain DRY. This seems to be pretty standard practice in dbt packages.
* If and when we do have to have distinct models for Postgres and Redshift, assuming the resulting table schema is the same, the change will not impact the package user. The model name will still be the same.

Comment (paulboocock) [reply dbt_project.yml:75]: Good points and certainly good enough for me.

**Pull request merged**